clang/unittests/ASTMatchers/ASTMatchersNarrowingTest.cpp

```diff
@@ -2251,6 +2251,14 @@ TEST_P(ASTMatchersTest, HasDependentName_DependentScopeDeclRefExpr) {
   EXPECT_TRUE(matches("template <typename T> struct S { static T foo(); };"
                       "template <typename T> void x() { S<T>::foo(); }",
                       dependentScopeDeclRefExpr(hasDependentName("foo"))));
+
+  EXPECT_TRUE(matches(
```

Comment:
nit: let's put this in a different test case called `HasDependentName_DependentNameType`

Reply:
Done, thank you